src/reactors/RelayOrderReactor.sol

```diff
@@ -53,109 +48,36 @@ contract RelayOrderReactor is ReactorEvents, ReactorErrors, ReentrancyGuard, IRe
             }
         }
 
-        _prepare(resolvedOrders);
-        _execute(resolvedOrders);
-        _fill(resolvedOrders);
-    }
-
-    function _execute(ResolvedRelayOrder[] memory orders) internal {
-        uint256 ordersLength = orders.length;
-        // actions are encoded as (address target, uint256 value, bytes data)[]
-        for (uint256 i = 0; i < ordersLength;) {
-            ResolvedRelayOrder memory order = orders[i];
-            uint256 actionsLength = order.actions.length;
-            for (uint256 j = 0; j < actionsLength;) {
-                (address target, uint256 value, bytes memory data) =
-                    abi.decode(order.actions[j], (address, uint256, bytes));
-                (bool success, bytes memory result) = target.call{value: value}(data);
-                if (!success) {
-                    // bubble up all errors, including custom errors which are encoded like functions
-                    assembly {
-                        revert(add(result, 0x20), mload(result))
-                    }
-                }
-                unchecked {
-                    j++;
-                }
-            }
-            unchecked {
-                i++;
-            }
-        }
+        _handleResolvedOrders(resolvedOrders);
     }
 
-    /// @notice validates and transfers input tokens in preparation for order fill
-    /// @param orders The orders to prepare
-    function _prepare(ResolvedRelayOrder[] memory orders) internal {
-        uint256 ordersLength = orders.length;
+    function _handleResolvedOrders(ResolvedRelayOrder[] memory orders) private {
         unchecked {
-            for (uint256 i = 0; i < ordersLength; i++) {
+            for (uint256 i = 0; i < orders.length; i++) {
                 ResolvedRelayOrder memory order = orders[i];
-
-                order.validate(msg.sender);
-
-                // Since relay order inputs specify recipients we don't pass in recipient here
-                transferInputTokens(order);
+                order.transferInputTokens(permit2);
+                order.executeActions(); // passing the whole order in? would be nice to not pass all the extra unrelated info in
```

Comment:
re. this comment, I don't mind this right now. While it would probably be cheaper if we moved `executeActions` back into the reactor instead of in the order lib, I like the pattern we have here